src/Compilers/Core/Portable/Syntax/SyntaxDiffer.cs

```diff
@@ -529,13 +529,26 @@ private void RecordDeleteOld(int oldNodeCount)
 
         private void RecordReplaceOldWithNew(int oldNodeCount, int newNodeCount)
         {
-            var oldSpan = GetSpan(_oldNodes, 0, oldNodeCount);
-            var removedNodes = CopyFirst(_oldNodes, oldNodeCount);
-            RemoveFirst(_oldNodes, oldNodeCount);
-            var newSpan = GetSpan(_newNodes, 0, newNodeCount);
-            var insertedNodes = CopyFirst(_newNodes, newNodeCount);
-            RemoveFirst(_newNodes, newNodeCount);
-            RecordChange(new ChangeRecord(new TextChangeRange(oldSpan, newSpan.Length), removedNodes, insertedNodes));
+            if (oldNodeCount == 1 && newNodeCount == 1)
+            {
+                var removedNode = _oldNodes.Pop();
```

Comment:
can you comment as much in the code to explain why this code should be kept around.